pkg/timelock/timelock.go

```diff
@@ -449,7 +451,9 @@ func (tw *Worker) handleLog(ctx context.Context, log types.Log) error {
 
 		if !isDone(ctx, tw.contract, cs.Id) && isOperation(ctx, tw.contract, cs.Id) {
 			tw.logger.Info().Hex(fieldTXHash, cs.Raw.TxHash[:]).Uint64(fieldBlockNumber, cs.Raw.BlockNumber).Msgf("%s received", eventCallScheduled)
-			tw.addToScheduler(cs)
+			if !tw.dryRun {
```

Comment:
I feel this changes the nature of the function, now a flag completely changes the output of it. 

The goal of the PR is having an option to avoid side effects right? That's not on the Worker side, but on the Scheduler side, so why don't we create another type of scheduler that doesn't execute? 
```go
func NewTimelockWorker(..., dryRun bool) (*Worker, error) {
    scheduler := *newScheduler(time.Duration(pollPeriod) * time.Second)
    if dryRun {
        // This could be exactly the same, except the execute, which does nothing
        scheduler := *newDryRunScheduler(time.Duration(pollPeriod) * time.Second)
    }
    tWorker := &Worker{
        ...
	scheduler:          scheduler
    }
    return tWorker, nil
}
```

Reply:
> The goal of the PR is having an option to avoid side effects right?

It's more about "testing the event listener". I'll give your suggestion a try but I have a feeling it will be harder to test and maintain in the end.

Reply:
The changes look good to me, how do you feel about it?